sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -852,6 +852,27 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] with a column dropped.
+   * @group dfops
+   */
+  def drop(col: String, cols: String*): DataFrame = {
```

Comment:
I just went with the way it is done in Pandas 
http://stackoverflow.com/questions/13411544/delete-column-from-pandas-dataframe

If that is not ok, I can update the PR